main/src/ca/uwaterloo/flix/language/phase/typer/TypeReduction.scala

```diff
@@ -293,13 +293,49 @@ object TypeReduction {
       case (Type.Apply(Type.Apply(Type.Cst(TypeConstructor.Array, _), elmType1, _), rcVar1, _),
       Type.Apply(Type.Apply(Type.Cst(TypeConstructor.Array, _), elmType2, _), rcVar2, _)) =>
         isSubtype(elmType1, elmType2)
+      // Arrow to Java function interface
+      // TODO INTEROP: generics support
+      case (Type.Apply(Type.Apply(Type.Apply(Type.Cst(TypeConstructor.Arrow(2), _), eff, _), var_arg, _), var_ret, _), Type.Cst(TypeConstructor.Native(clazz), _)) =>
+        (purifyType(var_arg), purifyType(var_ret)) match {
```

Comment:
it looks good now, but instead of purifyType, maybe just match on (var_arg, var_ret) and have case (Type.Cst(a, _), Type.Cst(b, _) => and then inside that case have the current match.

Reply:
Ah I see, something like this?
Thank you.